bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/things/GroupePSAHandler.java

```diff
@@ -0,0 +1,487 @@
+            final long pollingIntervalToUse = pollingIntervalM == null ? DEFAULT_POLLING_INTERVAL_M : pollingIntervalM;
+            groupepsaPollingJob = scheduler.scheduleWithFixedDelay(() -> pollStatus(), 1, pollingIntervalToUse * 60,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopGroupePSAPolling() {
+        ScheduledFuture<?> job = groupepsaPollingJob;
+        if (job != null) {
+            job.cancel(true);
+            groupepsaPollingJob = null;
+        }
+    }
+
+    private boolean isValidResult(VehicleStatus vehicle) {
+        return vehicle.getUpdatedAt() != null;
+    }
+
+    private boolean isConnected(VehicleStatus vehicle) {
+        Position lastPosition = vehicle.getLastPosition();
+        if (lastPosition == null) {
+            return false;
+        }
+
+        ZonedDateTime updatedAt = vehicle.getUpdatedAt();
+        if (updatedAt == null) {
+            return false;
+        }
+
+        return updatedAt.isAfter(ZonedDateTime.now().minusMinutes(onlineIntervalM));
+    }
+
+    private synchronized void updateGroupePSAState() {
+        if (System.nanoTime() - lastQueryTimeMs <= maxQueryFrequencyNanos) {
+            return;
+        }
+
+        lastQueryTimeMs = System.nanoTime();
+
+        String id = this.id;
+        if (id == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/conf-error-no-vehicle-id");
+            return;
+        }
+
+        GroupePSABridgeHandler groupepsaBridge = getBridgeHandler();
+        if (groupepsaBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/conf-error-no-bridge");
+            return;
+        }
+
+        try {
+            VehicleStatus vehicle = groupepsaBridge.getVehicleStatus(id);
+
+            if (vehicle != null && isValidResult(vehicle)) {
+                logger.trace("Vehicle: {}", vehicle.toString());
+
+                logger.debug("Update vehicle state now: {}, lastupdate: {}", ZonedDateTime.now(),
+                        vehicle.getUpdatedAt());
+
+                updateChannelState(vehicle);
+
+                if (isConnected(vehicle)) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                            "@text/comm-error-vehicle-not-connected-to-cloud");
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error-query-vehicle-failed");
+            }
+        } catch (GroupePSACommunicationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/comm-error-query-vehicle-failed");
+            logger.warn("Unable to query groupepsa status for:  {}. Error: {}", id, e.getMessage());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

You could include the exception's message in the thing status detail message to display it in the UI.